app/controllers/jams_controller.rb

```diff
@@ -4,7 +4,14 @@ class JamsController < ApplicationController
   # POST /rooms/:public_id/jams
   def create
     room = Room.find_by!(public_id: params[:room_id])
-    jam = room.jams.build(jam_params)
+    jam = room.jams.build(file: jam_params[:file])
+
+    # Set Jam tags
+    jam.bpm_list = jam_params[:bpm]
+    jam.song_key_list = jam_params[:song_key]
+    jam.jam_type_list = jam_params[:jam_type]
+    jam.style_list = jam_params[:style_list]
```

Comment:
that looks inefficient, should be somethings like ```
params[:tags].each { |context, list|  jam.set_tag_list_on(context, list) }
```
with the view that supports it in this generic way so we can just add any new context tags there without modifying the controller code.

Reply:
even better, to see if the plugin supports direct assignments, so we could just `room.jams.build(jam_params)` and it'd sort out tags on its own.